### PR TITLE
 Handle names ending with symbolic characters …

### LINKED ISSUE #202: Names end after _ ignoring symbols after the _
```scala
object defn {
  def Any_! = ???
}
defn.Any_!
```

### PR DESCRIPTION
**Pull request by nicolasstucki**
Fixes #202

### CODE REVIEW

File: src/typescript/Scala.tmLanguage.ts [hunk 1]
```diff
@@ -921,7 +921,7 @@ export const scalaTmLanguage: TmLanguage = {
       comment: 'For themes: Brackets look nice when colored.'
     },
     qualifiedClassName: {
-      match: '(\\b([A-Z][\\w]*))',
+      match: `(\\b([A-Z][\\w]*)(?:(?<=_)${opchar}+)?)`,
```
Comment (MaximeKjaer) [inline src/typescript/Scala.tmLanguage.ts:924]: Could there be multiple `_` or is only one allowed?

**Review by MaximeKjaer — APPROVED** [review]
Looks fine to me. Should we perhaps rebase to avoid having the onigurama upgrade twice? Or I suppose things will look okay in the merge commit?

File: src/typescript/Scala.tmLanguage.ts (hunk 1 above)
Comment (nicolasstucki) [reply src/typescript/Scala.tmLanguage.ts:924]: The scanner only allows one `_`. 

See https://github.com/lampepfl/dotty/blob/master/compiler/src/dotty/tools/dotc/parsing/Scanners.scala#L1034

Comment (MaximeKjaer) [reply src/typescript/Scala.tmLanguage.ts:924]: Good, let's merge then!

**Comment by Psoffie009** [comment]
> Fixes #202

**Comment by Psoffie009** [comment]
> Looks fine to me. Should we perhaps rebase to avoid having the onigurama upgrade twice? Or I suppose things will look okay in the merge commit?

**Pull request merged**